microprofile-jwt/src/main/java/org/jboss/eap/qe/microprofile/jwt/tools/RsaKeyTool.java

```diff
@@ -0,0 +1,132 @@
+package org.jboss.eap.qe.microprofile.jwt.tools;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.net.URI;
+import java.security.KeyFactory;
+import java.security.NoSuchAlgorithmException;
+import java.security.PrivateKey;
+import java.security.interfaces.RSAPrivateCrtKey;
+import java.security.interfaces.RSAPublicKey;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.PKCS8EncodedKeySpec;
+import java.security.spec.RSAPublicKeySpec;
+import java.util.Arrays;
+import java.util.Base64;
+import javax.json.Json;
+import javax.json.JsonObject;
+
+/**
+ * Utility class for this test suite to return a RSA key in various formats (such as JWK, PEM, etc).
+ * <p>
+ * An instance of this class will prepare one single RSA key pair that is reused throughout the lifetime of the instance
+ * and every method invocation will operate on the same key pair. In addition to the key pair, a key identifier ("kid")
+ * for use with JWK will be generated.
+ * <p>
+ *
+ * @author Daniel Pfeifer
```

Comment:
Is it ok to copy this class ?

Reply:
It is part Thorntail testsuite right now with little bit different API.